src/rust/engine/src/nodes.rs

```diff
@@ -356,7 +372,7 @@ impl Select {
             product: self.product().clone(),
             variants: self.variants.clone(),
             task: task,
-            entry: entry.clone(),
+            entry: Arc::new(entry.clone()),
```

Comment:
Seems a little strange for entry to be both `Arc`'d _and_ `clone`'d here. My instinct (which may be wrong) is that either it's a cheap to clone type, and we shouldn't be `Arc`ing it, or it's an expensive to clone type, and we should always be `Arc`ing it...

Reply:
Yea, I have a separate review that attempts to Arc up the rule-graph, since it is full of tiny `Vec`s. On the other hand, it does not appear to impact performance in a significant way... so perhaps this would not either. It's very surprising to me that all of the allocation doesn't have a noticeable impact.